install/first_google_boot.sh

```diff
@@ -224,11 +224,22 @@ function extract_spinnaker_kube_credentials() {
   chown -R spinnaker:spinnaker $(dirname $config_path)
 
   local kube_cluster=$(get_instance_metadata_attribute "kube_cluster")
+  local kube_project=$(get_instance_metadata_attribute "kube_project")
+  local kube_zone=$(get_instance_metadata_attribute "kube_zone")
+
+  if [ -z "$kube_project" ]; then
+    kube_project=$MY_PROJECT
+  fi
+
+  if [ -z "$kube_zone" ]; then
+    kube_zone=$MY_ZONE
+  fi
+
   if [ -n "$kube_cluster" ]; then
```

Comment:
Well these aren't fields that can be supplied to spinnaker.yml, since they're needed before spinnaker.yml is ever read/interpreted, so that's not an option here unfortunately.

Also, the "immutable infrastructure" vision we are pushing encourages trashing VMs, even for config changes, so I'm not too worried about that.

From a C2D perspective, presenting the fact that the cluster needs to exist in the supplied project & zone seems to justify failing early.